pallets/marketplace/src/lib.rs

```diff
@@ -213,24 +255,109 @@ pub mod pallet {
         pub fn delete_advertisement(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
             let who = ensure_signed(origin)?;
 
-            let ad =
-                <StoredAdvertisement<T>>::get(&who).ok_or(Error::<T>::AdvertisementNotFound)?;
+            <StoredAdvertisementRestriction<T>>::get(&who)
```

Comment:
This is doing nothing. The deletion in line 269 should handle this check.

Reply:
It's checking if the advertisement actually exists (because `<StoredAdvertisementRestriction<T>>::remove(&who)` below does not fail when not exist)